### PR TITLE
Redirect stdout when checking imports.

### PR DESCRIPTION
**Pull request by aptenodytes-forsteri**
Fixes https://github.com/bazelbuild/rules_python/issues/1006

Also, set PYTHONNOUSERSITE so that the script doesn't even look in site packages when checking modules.

Fix typo with capitilize.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #1006 

## What is the new behavior?
```echo "pygame" | bazel run //gazelle:std_modules
false
```
No extraneous output.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: gazelle/std_modules.py [hunk 1]
```diff
@@ -3,30 +3,27 @@
...
+import os
 import sys
-
-
-# Don't return any paths, all userland site-packages should be ignored.
-def __override_getusersitepackages__():
-    return ""
-
-
-site.getusersitepackages = __override_getusersitepackages__
```
Comment (aptenodytes-forsteri) [inline gazelle/std_modules.py:15]: This code wasn't having the desired effect.

**Comment by aptenodytes-forsteri** [comment]
I might be able to add a test case

**Comment by aptenodytes-forsteri** [comment]
> I might be able to add a test case

Scratch that, looks a bit more involved because. Probably enough of an edge case here that it doesn't really need to be covered by a test.

File: gazelle/std_modules.py [hunk 2]
```diff
@@ -3,30 +3,28 @@
...
+            return True
+        except Exception:
+            return False
 
 
 def main(stdin, stdout):
     for module in stdin:
         module = module.strip()
-        # Don't print the boolean directly as it is captilized in Python.
+        # Don't print the boolean directly as it is capitalized in Python.
```
Comment (f0rmiga) [inline gazelle/std_modules.py:27]: This was the Brazilian spelling. 🤣 
Thanks for fixing it.

File: gazelle/std_modules.py (hunk 1 above)
Comment (f0rmiga) [reply gazelle/std_modules.py:15]: Interesting.

**Review by f0rmiga — APPROVED** [review]
